spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -810,4 +823,123 @@ private Catalog.TableBuilder newBuilder(Identifier ident, Schema schema) {
   public Catalog icebergCatalog() {
     return icebergCatalog;
   }
+
+  @Override
+  public Identifier[] listViews(String... namespace) {
+    if (null != asViewCatalog) {
+      return asViewCatalog.listViews(Namespace.of(namespace)).stream()
+          .map(ident -> Identifier.of(ident.namespace().levels(), ident.name()))
+          .toArray(Identifier[]::new);
+    }
+
+    return new Identifier[0];
+  }
+
+  @Override
+  public View loadView(Identifier ident) throws NoSuchViewException {
+    if (null != asViewCatalog) {
+      try {
+        org.apache.iceberg.view.View view = asViewCatalog.loadView(buildIdentifier(ident));
+        return new SparkView(catalogName, view);
+      } catch (org.apache.iceberg.exceptions.NoSuchViewException e) {
+        throw new NoSuchViewException(ident);
+      }
+    }
+
+    throw new NoSuchViewException(ident);
+  }
+
+  @Override
+  public View createView(
```

Comment:
Minor: Can you replace these implementations with `throw UnsupportedOperationException(...)`?

This code is untested so it would be better to get this PR in and implement the methods as we add tests.

Reply:
ok I've replaced all methods to throw an UOE that aren't exercised by the tests in this PR